compositor_common/src/frame.rs

```diff
@@ -1,15 +1,48 @@
-use std::time::Duration;
+use std::{collections::HashMap, sync::Arc, time::Duration};
 
 use crate::scene::Resolution;
 
-#[allow(dead_code)]
+// Clone is temporary. Testing use only.
 #[derive(Debug, Clone)]
 pub struct Frame {
     pub data: YuvData,
     pub resolution: Resolution,
     pub pts: Duration,
 }
 
+pub type Pts = Duration;
+pub type InputID = u32;
+
+/// TODO: This should be a rational.
+#[derive(Debug, Clone, Copy)]
+pub struct Framerate(pub u32);
+
+impl Framerate {
+    pub fn get_interval_duration(self) -> Duration {
+        Duration::from_nanos((1_000_000_000 / self.0).into())
+    }
+}
+
+#[derive(Debug)]
+pub struct FramesBatch {
```

Comment:
I think a common crate should describe types that are used everywhere, but this value describes queue output and renderer input, so I think it should be defined in the renderer crate.

Also, the name is technically correct, but I feel it's a bit misleading, when I hear batch I understand there would be a list of consecutive frames for single input "batched" together. Maybe consider InputFrames, FrameSet or maybe using rust naming InstantFrameSet?